File: src/labelDetector.js

async function detectLabels(path) {
    const vision = require('@google-cloud/vision');

    const client = new vision.ImageAnnotatorClient();
    const [result] = await client.labelDetection(path);
    const labels = result.labelAnnotations;

    let boiled = false;
    labels.forEach(label => {
        if(label.description.includes('Boiling')) {
            boiled = true;
        }
    });
    let str = boiled ? 'boiling' : 'not boiling';
    console.log('Your water is ' + str)
    return boiled;
}

detectLabels('C:\\Users\\arnav\\OneDrive - University of Florida\\whack-ding\\resources\\water9.jpg');

File: src/labelDetector.py
import io
import os
from google.cloud import vision

def detect_labels(path):
    """Detects labels in the file."""
    client = vision.ImageAnnotatorClient()

    with io.open(path, 'rb') as image_file:
        content = image_file.read()

    image = vision.Image(content=content)

    response = client.label_detection(image=image)
    labels = response.label_annotations
    print('Labels:')
    print(labels)
    # for label in labels:
    #     print(label.description)

    if response.error.message:
        raise Exception(
            '{}\nFor more info on error messages, check: '
            'https://cloud.google.com/apis/design/errors'.format(
                response.error.message))


detect_labels('C:\\Users\\arnav\\OneDrive - University of Florida\\whack-ding\\resources\\wakeupcat.jpg')
